docs/source/user_guide/guide.rst

```diff
@@ -0,0 +1,497 @@
+
+    **info dict:**
+
+    +----------+------------+-----------------------------+
+    | dict key | value type | value description           |
+    +==========+============+=============================+
+    | value    | np.ndarray | new [min, max] of selection |
+    +----------+------------+-----------------------------+
+
+`ImageWidget`
+-------------
+
+Often times, developing UIs for interacting with multi-dimension image data can be tedious and repetitive.
+In order to aid with common image and video visualization requirements the `ImageWidget` automatically generates sliders
+to easily navigate through different dimensions of your data. Let's look at an example: ::
+
+    import fastplotlib as fpl
+    import imageio.v3 as iio
+
+    movie = iio.imread("imageio:cockatoo.mp4")
+
+    # convert RGB movie to grayscale
+    gray_movie = np.dot(movie[..., :3], [0.299, 0.587, 0.114])
+
+    iw_movie = ImageWidget(
+    data=gray_movie,
+    cmap="gray"
+    )
+
+    iw_movie.show()
+
+.. image:: _static/guide_image_widget.gif
+
+Animations
+----------
+
+An animation function is a user-defined function that gets called on every rendering cycle. Let's look at an example: ::
+
+    import fastplotlib as fpl
+    import numpy as np
+
+    data = np.random.rand(512, 512)
+
+    fig = fpl.Figure()
+
+    fig[0,0].add_image(data=data, name="random-img")
+
+    def update_data(plot_instance):
+        new_data = np.random.rand(512, 512)
+        plot_instance["random-img"].data = new_data
+
+    fig[0,0].add_animations(update_data)
+
+    fig.show()
+
+.. image:: _static/guide_animation.gif
+
+Here we are defining a function that updates the data of the `ImageGraphic` in the plot with new random data. When adding an animation function, the
+user-defined function will receive a plot instance as an argument when it is called.
+
+Spaces
+------
+
+There are several spaces to consider when using `fastplotlib`:
+
+1) World Space
+
+    World space is the 3D space in which objects live. World space has no limits on its size. Objects
+    and the camera can exist anywhere in this space. Objects in this space exist relative to a larger
+    world.
+
+2) Data Space
+
+    Data space is simply the world space plus any offset or rotation that has been applied to an object.
+
+3) Screen Space
+
+    Screen space is a 2D space represented in pixels. This space is constrained by the screen width and screen height.
+    In the rendering process, the camera is responsible for projecting the world space into screen space.
+
+.. note::
+    When interacting with `Graphic` objects, there is a very helpful function for mapping screen space to world space
+    (`Figure.map_screen_to_world(pos=(x, y))`). This can be particularly useful when working with click events where click
+    positions are returned in screen space but `Graphic` objects that you may want to interact with exist in world
+    space.
+
+
+Using `fastplotlib` interactively
```

Comment:
This section needs more clarity, add subsections for each of jupyter and ipython. In jupyter they can use jupyter-rfb, qt, or glfw. In ipython they can use only qt or glfw.